src/backend/core/api/viewsets.py

```diff
@@ -750,17 +820,39 @@ def perform_create(self, serializer):
             self.request.user,
         )
 
+    def perform_update(self, serializer):
+        """Update an access to the document and notify the collaboration server."""
+        access = serializer.save()
+
+        access_user_id = None
+        if access.user:
+            access_user_id = str(access.user.id)
+
+        # Notify collaboration server about the access change
+        CollaborationService().reset_connections(
+            str(access.document.id), access_user_id
+        )
+
+    def perform_destroy(self, instance):
+        """Delete an access to the document and notify the collaboration server."""
+        instance.delete()
+
+        # Notify collaboration server about the access removed
+        CollaborationService().reset_connections(
+            str(instance.document.id), str(instance.user.id)
+        )
+
```

Comment:
a signal directly on the DocumentAccess model is better place to do that because the access could be changed from elsewhere.

Reply:
I think I will do it in another PR if it is ok. It asks to modify lot of tests in the modals.